android/expoview/src/main/java/host/exp/exponent/kernel/Kernel.java

```diff
@@ -569,6 +574,11 @@ public void handleIntent(Activity activity, Intent intent) {
     openDefaultUrl();
   }
 
+  // Certain links should just open the HomeScreen
+  private boolean shouldOpenUrl(@NonNull Uri uri) {
+    return !uri.toString().contains("expo.io/expo-go");
```

Comment:
.contains checks in URLs are often unsafe - what about "notexpo.io/expo-go-be-vulnerable"?

Reply:
hm yeah I can change this to `.equals`. Only specified domains (in AndroidManifest) can reach this, but the argument can still be made for "expo.io/expo-go-lol"